letta/server/server.py

```diff
@@ -1955,24 +1955,35 @@ def run_tool_from_source(
             if sandbox_run_result is None:
                 raise ValueError(f"Tool with id {tool.id} returned execution with None")
             function_response = str(sandbox_run_result.func_return)
+            stdout = [s for s in sandbox_run_result.stdout if s.strip()]
+            stderr = [s for s in sandbox_run_result.stderr if s.strip()]
+
+            # expected error
+            if stderr:
+                error_msg = self.get_error_msg_for_func_return(tool.name, stderr[-1])
+                return FunctionReturn(
+                    id="null",
```

Comment:
nit: code style around setting id/function_call_id to predefined string - ideally, we would have maybe a child type of FunctionReturn without id/function_call_id

Reply:
Okay to leave todo :)